app/controllers/suggested_edit_controller.rb

```diff
@@ -0,0 +1,98 @@
+class SuggestedEditController < ApplicationController
+  before_action :set_suggested_edit, only: [:show, :approve, :reject]
+
+  def show
+    render layout: 'without_sidebar'
+  end
+
+  def approve
+    return unless @edit.active?
+
+    @post = @edit.post
+    unless check_your_privilege('Edit', @post, false)
+      render(json: { status: 'error', redirect_url: 'You need the Edit privilege to approve edits' }, status: 400)
+
+      return
+    end
+
+    PostHistory.post_edited(@post, @edit.user, before: @post.body_markdown,
+                                               after: @edit.body_markdown, comment: params[:edit_comment])
+
+    if @post.update(applied_details)
+      @edit.update(active: false, accepted: true, rejected_comment: '', decided_at: DateTime.now,
+                                                  decided_by: current_user, updated_at: DateTime.now)
+      flash[:success] = 'Edit approved successfully.'
+      if @post.question?
+        render(json: { status: 'success', redirect_url: url_for(controller: :posts, action: :share_q,
+                                                                id: @post.id) }, status: 200)
+
+        return
+      elsif @post.answer?
+        render(json: { status: 'success', redirect_url: url_for(controller: :posts, action: :share_a,
+                                                        qid: @post.parent.id, id: @post.id) }, status: 200)
```

Comment:
No harm in it, but you can omit the `status: 200` here if you want since that's the default status.